project/src/components/login-form/login-form.tsx

```diff
@@ -30,39 +30,22 @@ const LoginForm = () => {
 
   const dispatch = useAppDispatch();
 
-  const isFieldsGroupValid = () => formData.email.error || formData.password.error;
+  const isFieldsGroupValid = Object.values(formData)
+    .map((value) => value.error)
+    .every((elem) => !elem);
 
   const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
     const { name, value } = e.target;
+    const isValid = (inputValue: string, matchedValue: RegExp) => matchedValue.test(inputValue);
```

Comment:
Да в целом можно просто без функции 
```suggestion
    const isValid = rule.test(value);
```